src/Features/VolumetricLighting.cpp

```diff
@@ -66,15 +159,83 @@ void VolumetricLighting::DataLoaded()
 void VolumetricLighting::PostPostLoad()
 {
 	if (REL::Module::IsVR()) {
-		if (settings.EnabledVL)
+		if (settings.ExteriorEnabled || settings.InteriorEnabled)
 			EnableBooleanSettings(hiddenVRSettings, GetName());
 		auto address = REL::RelocationID(100475, 0).address() + 0x45b;  // AE not needed, VR only hook
 		logger::info("[{}] Hooking CopyResource at {:x}", GetName(), address);
 		REL::safe_fill(address, REL::NOP, 7);
 		stl::write_thunk_call<CopyResource>(address);
+
+		// Skip volumetric lighting rendering
+		REL::safe_write(REL::RelocationID(35560, 0).address() + REL::Relocate(0x254, 0), &REL::JMP8, 1);
+		// Move it to render after depth to ensure camera matches rest of scene
+		stl::write_thunk_call<RenderDepth>(REL::RelocationID(35560, 0).address() + REL::Relocate(0x2EE, 0));
 	}
+
+	bEnableVolumetricLighting = reinterpret_cast<bool*>(REL::VariantOffset(0x3232EF0, 0x338C544, 0x3485362).address());
+	gVolumetricLightingSizeLow = reinterpret_cast<TextureSize*>(REL::VariantOffset(0x3233090, 0x338C550, 0x3485630).address());
+	gVolumetricLightingSizeMedium = reinterpret_cast<TextureSize*>(REL::VariantOffset(0x323309C, 0x338C55C, 0x348563C).address());
+	gVolumetricLightingSizeHigh = reinterpret_cast<TextureSize*>(REL::VariantOffset(0x32330A8, 0x338C568, 0x3485648).address());
+	defaultSizeHigh = *gVolumetricLightingSizeHigh;
+}
+
+void VolumetricLighting::EarlyPrepass()
+{
+	const auto interiorCell = RE::TES::GetSingleton()->interiorCell;
```

Comment:
Would this be better as a global? Or at least TES?

Reply:
Good idea, swapped to the global